### PR TITLE
black: specify major version 22 only

### PR DESCRIPTION
**Pull request by Minnozz**
This changes some places where black (the Python formatter) is referenced:
- The GitHub Action version itself is set to `stable` (as recommended)
- The GitHub Action version constraint for black is relaxed to `22.*` (formatting is guaranteed to be stable within a major version)
- The required black major version is set in `pyproject.toml` to avoid formatting with a different major version. This improves consistency between different contributors (eg, me) having different black versions set up in their IDE
- Add black as a dependency so IDEs (like VSCode) can be configured to use black from the local environment

### CODE REVIEW

**Comment by Minnozz** [comment]
I'm not well-versed enough with Python development setups to know if the last point (adding black as a dev dependency) is acceptable. Input is welcome!

File: requirements.txt [hunk 1]
```diff
@@ -41,6 +41,7 @@ setuptools>=65.5.1 # Not a direct dependency, pinned to get a security fix
 tornado==6.3.3 # Not a direct dependency, pinned to get a security fix
 
 # Dev
+black==22.*
```
Comment (dato) [inline requirements.txt:44]: Would it make sense to update dev-tools/requirements.txt with the same wildcard?

**Review by dato — APPROVED** [review]
> I'm not well-versed enough with Python development setups to know if the last point (adding black as a dependency) is acceptable. Input is welcome!

Sounds acceptable to me, we just need to drop it in the merge to the _production_ branch (i.e. when resolving conflicts there).

File: requirements.txt (hunk 1 above)
Comment (Minnozz) [reply requirements.txt:44]: Definitely, I forgot that place

**Comment by Minnozz** [comment]
> Sounds acceptable to me, we just need to drop it in the merge to the _production_ branch (i.e. when resolving conflicts there).

Cool! (I wasn't aware a production branch existed until now)

**Pull request merged**